### PR TITLE
Removed dependency of openssl.ripemd160  and replaced it by embit dependency

### PR DESCRIPTION
**Pull request by relativisticelectron**
Removed dependency of openssl.ripemd160  and replaced it by embit dependency

This is useful because embit handles the missing ripemd160 in openssl3: https://github.com/diybitcoinhardware/embit/issues/28 starting from version 0.4.13

see also: https://github.com/cryptoadvance/specter-desktop/pull/1688#issuecomment-1116169695

### CODE REVIEW

**Comment by relativisticelectron** [comment]
@k9ert : Could you review?

File: src/cryptoadvance/specter/devices/specter.py [hunk 1]
```diff
@@ -173,6 +174,5 @@ def get_wallet_fingerprint(wallet):
     Unique fingerprint of the wallet -
     first 4 bytes of hash160 of its descriptor
     """
-    h256 = hashlib.sha256(get_wallet_qr_descriptor(wallet).encode()).digest()
-    h160 = hashlib.new("ripemd160", h256).digest()
+    h160 = hash160(get_wallet_qr_descriptor(wallet).encode()).digest()
```
Comment (stepansnigirev) [inline src/cryptoadvance/specter/devices/specter.py:177]: hash160 from embit doesn't need digest()

```suggestion
    h160 = hash160(get_wallet_qr_descriptor(wallet).encode())
```

Comment (relativisticelectron) [reply src/cryptoadvance/specter/devices/specter.py]: Thanks. I missed that.

**Comment by k9ert** [comment]
I've created a test for this and tried to improve the test-framework along the way. If test goes through, happy to merge but please, @relativisticelectron have a look before as well.

File: tests/test_devices_specter.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        "someName", 1, "wpkh", [specter_device.keys[0]], [specter_device]
+    )
+
+    assert wallet != None
+
+    assert (
+        get_wallet_qr_descriptor(wallet)
+        == "wpkh([ccf2e5c3/84h/1h/0h]tpubDCnYSFavtHxX7w8S8GyYwQ2bQPeA5fSVd2WqFzY7BeE1DKhqvq9Qdyz4AM13xGQvo1J5c46ixbW84evZhqerR3eh1r2tndT8r51p3sxiQ8F)"
+    )
+    assert get_wallet_fingerprint(wallet) == ""
```
Comment (moneymanolis) [inline tests/test_devices_specter.py]: Why assert empty strings here?

Comment (k9ert) [reply tests/test_devices_specter.py]: That was before [e6f1680](https://github.com/cryptoadvance/specter-desktop/pull/1702/commits/e6f168080cf290d81f5e6a8433b09270dd1dbee0) fixed the output and i didn't know what to expect.
You caught me, a lot of my tests are effectively just regression-tests but that's more than often good enough.

**Pull request merged**